jvm/src/common/Language/Java/Internal.hs

```diff
@@ -161,14 +162,18 @@ getStaticFieldAsJValue retsing cname fname = do
     SVoid -> fail "getStaticField cannot yield an object of type void"
     _ -> JObject <$> getStaticObjectField klass field
 
+-- | The maximum supported number of arguments to variadic functions.
+maxVariadicArgs :: Int
+maxVariadicArgs = 32
```

Comment:
Nitpick: Is it hard to get the constant used at the type level in the `TypeError` message that refers to it? :smiling_imp:

Reply:
Well, you could have

```
type MaxVariadicArgs = 32

maxVariadicArgs :: Int
maxVariadicArgs = fromInteger $ natVal @MaxVariadicArgs Proxy
```

Gee, it would be nice to have dependent types... :)